.github/workflows/testing.yml

```diff
@@ -0,0 +1,97 @@
+name: Testing
+
+# Trigger on pushes, PRs (excluding documentation changes), and nightly.
+on:
+  push:
+  pull_request:
+    paths-ignore:
+      - 'Documentation/**'
+      - 'version.go'
+  schedule:
+    - cron: 0 0 * * * # daily at 00:00
+
+# Always force the use of Go modules
+env:
+  GO111MODULE: on
+
+jobs:
+  # Check generated protos match their source repos (optional for PRs).
+  vet-proto:
+    runs-on: ubuntu-latest
+    steps:
+      # Setup the environment.
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.14
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      # Run the vet checks.
+      - name: vet
+        run: ./vet.sh -install && ./vet.sh
```

Comment:
This workflow runs vet (gofmt/golint...) twice now.
Should we move the proto regen part out to it's own job?

Reply:
I was thinking as a follow-up change I would add an option to vet to make it ONLY run the proto checks.  WDYT?

Reply:
SG.

Do the proto checks need anything from vet.sh? Or just move it to a separate `.sh`

Reply:
They both need the checks up front to make sure your branch is clean & revert any changes it makes to it.

I would probably keep them together to support the "running offline" use-case better.  `VET_ONLY_PROTO=1` or something?